nilearn/plotting/html_stat_map.py

```diff
@@ -1,148 +1,438 @@
+        The colormap for specified image. The colormap *must* be
+        symmetrical.
+    symmetric_cbar : boolean or 'auto' (default='auto')
+        Specifies whether the colorbar should range from -vmax to vmax
+        or from vmin to vmax. Setting to 'auto' will select the latter if
+        the range of the whole image is either positive or negative.
+        Note: The colormap will always be set to range from -vmax to vmax.
+    dim : float, 'auto' (default='auto')
+        Dimming factor applied to background image. By default, automatic
+        heuristics are applied based upon the background image intensity.
+        Accepted float values, where a typical scan is between -2 and 2
+        (-2 = increase constrast; 2 = decrease contrast), but larger values
+        can be used for a more pronounced effect. 0 means no dimming.
+    vmax : float, or None (default=)
+        max value for mapping colors.
+    resampling_interpolation : string, optional (default nearest)
+        The interpolation method for resampling
+        See nilearn.image.resample_img
+    opacity : float in [0,1] (default 1)
+        The level of opacity of the overlay (0: transparent, 1: opaque)
 
     Returns
     -------
-    StatMapView : plot of the stat map.
+    view : the viewer object.
         It can be saved as an html page or rendered (transparently) by the
         Jupyter notebook.
-
     """
-    stat_map_img = check_niimg_3d(stat_map_img, dtype='auto')
-    if bg_img == 'MNI152':
-        bg_img = datasets.load_mni152_template()
-    if bg_img is not None and bg_img is not False:
-        bg_img = image.load_img(bg_img)
-        stat_map_img = image.resample_to_img(stat_map_img, bg_img)
-        bg_mask = image.new_img_like(bg_img, bg_img.get_data() != 0)
-        stat_map_img = image.new_img_like(
-            stat_map_img, stat_map_img.get_data() * bg_mask.get_data())
-        encoded_bg = '"{}"'.format(_encode_nii(bg_img))
+
+    cmap = plt.cm.get_cmap(cmap)
+
+    # Mask stat map
+    mask_img, stat_map_img, data = _mask_stat_map(stat_map_img,threshold)
+
+    # Get color bar and data ranges
+    cbar_vmin, cbar_vmax, vmin, vmax = _get_colorbar_and_data_ranges(
+        data, vmax, symmetric_cbar, kwargs)
+
+    # Load background image
+    bg_img, bg_min, bg_max, black_bg = _load_bg_img(stat_map_img, bg_img,
+                                                    black_bg, dim)
+
+    # Resample stat map
+    stat_map_img, mask_img = _resample_stat_map(stat_map_img, bg_img, mask_img,
+                                            resampling_interpolation)
+
+    # Get the slices for the cut
+    cut_slices = _get_cut_slices(stat_map_img, cut_coords, threshold)
+
+    # Create a base64 sprite for the background
+    bg_sprite = BytesIO()
+    bg_data = _safe_get_data(bg_img, ensure_finite=True)
+    _save_sprite(bg_data, bg_sprite, bg_max, bg_min, None, 'gray', 'png')
+    bg_base64 = _bytesIO_to_base64(bg_sprite)
```

Comment:
I would rather have the _bytesIO_to_base64 function inlined. The reason is that it is passed an open file handle and closes it. This is not very good behavior for a function.